include/boost/hana/config.hpp

```diff
@@ -72,7 +97,9 @@ Distributed under the Boost Software License, Version 1.0.
 //////////////////////////////////////////////////////////////////////////////
 #if (__cplusplus < 201400)
 #   if defined(_MSC_VER)
+#if _MSC_VER < 1915
```

Comment:
Then it will pick the #else part which is for GNU compiler :-)

The first #if is to determine how to emit the warning (via #pragma message or #warning). The second #if is to determine whether to emit the warning.

Reply:
I considered that option too. However, we discovered that various libraries don't work with this flag (either compiler bugs or source issues), so people may choose not to use this flag and I prefer to support boost.hana even if this flag is not used.

I don't have strong preference, though.